osu.Game/Online/Chat/ChatManager.cs

```diff
@@ -0,0 +1,265 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using osu.Framework.Configuration;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Logging;
+using osu.Framework.Threading;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+
+namespace osu.Game.Online.Chat
+{
+    /// <summary>
+    /// Manages everything chat related
+    /// </summary>
+    public sealed class ChatManager : IOnlineComponent

```

Comment:
Should also be a `Component`, and added to the draw heirarchy (see `APIAccess`). You won't need to pass a `Scheduler` in any more.